core/trino-main/src/main/java/io/trino/operator/StreamingExchangeClientBuffer.java

```diff
@@ -143,6 +145,12 @@ public synchronized void taskFailed(TaskId taskId, Throwable t)
             return;
         }
         checkState(activeTasks.contains(taskId), "taskId not registered: %s", taskId);
+
+        if (t instanceof TrinoException && ((TrinoException) t).getErrorCode() == REMOTE_TASK_FAILED.toErrorCode()) {
+            // let coordinator handle this
```

Comment:
Lost on this one. How does the coordinator get this information?

Reply:
We want the coordinator discover this failure to propagate the correct error information instead of failing a downstream task.